packages/path_provider/path_provider_foundation/lib/path_provider_foundation.dart

```diff
@@ -2,106 +2,8 @@
-
-  /// Registers this class as the default instance of [PathProviderPlatform]
-  static void registerWith() {
-    PathProviderPlatform.instance = PathProviderFoundation();
-  }
-
-  @override
-  Future<String?> getTemporaryPath() {
-    return _pathProvider.getDirectoryPath(DirectoryType.temp);
-  }
-
-  @override
-  Future<String?> getApplicationSupportPath() async {
-    final String? path =
-        await _pathProvider.getDirectoryPath(DirectoryType.applicationSupport);
-    if (path != null) {
-      // Ensure the directory exists before returning it, for consistency with
-      // other platforms.
-      await Directory(path).create(recursive: true);
-    }
-    return path;
-  }
-
-  @override
-  Future<String?> getLibraryPath() {
-    return _pathProvider.getDirectoryPath(DirectoryType.library);
-  }
-
-  @override
-  Future<String?> getApplicationDocumentsPath() {
-    return _pathProvider.getDirectoryPath(DirectoryType.applicationDocuments);
-  }
-
-  @override
-  Future<String?> getApplicationCachePath() async {
-    final String? path =
-        await _pathProvider.getDirectoryPath(DirectoryType.applicationCache);
-    if (path != null) {
-      // Ensure the directory exists before returning it, for consistency with
-      // other platforms.
-      await Directory(path).create(recursive: true);
-    }
-    return path;
-  }
-
-  @override
-  Future<String?> getExternalStoragePath() async {
-    throw UnsupportedError(
-        'getExternalStoragePath is not supported on this platform');
-  }
-
-  @override
-  Future<List<String>?> getExternalCachePaths() async {
-    throw UnsupportedError(
-        'getExternalCachePaths is not supported on this platform');
-  }
-
-  @override
-  Future<List<String>?> getExternalStoragePaths({
-    StorageDirectory? type,
-  }) async {
-    throw UnsupportedError(
-        'getExternalStoragePaths is not supported on this platform');
-  }
-
-  @override
-  Future<String?> getDownloadsPath() {
-    return _pathProvider.getDirectoryPath(DirectoryType.downloads);
-  }
-
-  /// Returns the path to the container of the specified App Group.
-  /// This is only supported for iOS.
-  Future<String?> getContainerPath({required String appGroupIdentifier}) async {
-    if (!_platformProvider.isIOS) {
-      throw UnsupportedError(
-          'getContainerPath is not supported on this platform');
-    }
-    return _pathProvider.getContainerPath(appGroupIdentifier);
-  }
-}
-
-/// Helper class for returning information about the current platform.
-@visibleForTesting
-class PathProviderPlatformProvider {
-  /// Specifies whether the current platform is iOS.
-  bool get isIOS => Platform.isIOS;
-}
+// path_provider_foundation is implemented using FFI; export a stub for
+// platforms that don't support FFI (e.g., web) to avoid having transitive
+// dependencies break web compilation.
+export 'src/path_provider_foundation_stub.dart'
+    if (dart.library.ffi) 'src/path_provider_foundation_real.dart';
```

Comment:
What are your thoughts on remove `_real`? Adding `_stub` to the placeholder one seems sufficient to me

```suggestion
    if (dart.library.ffi) 'src/path_provider_foundation.dart';
```

Reply:
This is how `path_provider_windows` is set up, so I mirrored it. IIRC we did this to avoid having two files with the same name in the same package? I don't have strong feelings about it though.